### ISSUE TITLE
Dokka HTML: optionally add a custom link to source code homepage in navbar

### ISSUE THREAD

**Issue by aSemy** [Kotlin/dokka#2948]
**Is your feature request related to a problem? Please describe**

I would like to be able to add a link to my source code homepage in the Dokka HTML nav bar

Example: a link to GitHub

<img width="291" alt="image" src="https://user-images.githubusercontent.com/897017/228296879-f3e6db41-420c-439e-b087-06291a3a891e.png">

https://adamko-dev.github.io/dokkatoo/

**Describe the solution you'd like**

Update the navbar template to include an optional 'homepage' icon. If a URL is provided, the icon is shown. I would also like to be able to theme the icon using CSS. 

Ideally, Dokka would pre-package some icons so that I could select 'GitHub', 'GitLab', 'Gitea', etc as an icon.

**Describe alternatives you've considered**

It is not possible to do this via CSS `:before`/`:after`, as pseudo elements cannot have href links.

I have manually implemented this using find-replace in Dokkatoo

https://github.com/adamko-dev/dokkatoo/blob/5da6e4d1ed8654b9c6b5dae7ece6a440bea8230c/docs/build.gradle.kts#L27-L57

```kotlin
tasks.dokkatooGeneratePublicationHtml {
  doLast {
    outputDirectory.get().asFile.walk()
      .filter { it.isFile && it.extension == "html" }
      .forEach { file ->
        file.writeText(
          file.readText()
            .replace(
              """<html>""",
              """<html lang="en">""",
            )
            .replace(
              """
                <button id="theme-toggle-button">
              """.trimIndent(),
              """
                <div id="github-link"><a href="https://github.com/adamko-dev/dokkatoo/"></a></div>
                <button id="theme-toggle-button">
              """.trimIndent(),
            )
        )
      }
  }
}
```

**Additional context**

**Are you willing to provide a PR?**

Not right now - the find and replace workaround is suitable.

**Comment by IgnatBeresnev** [comment]
It would be beyond awesome to add the source links icons to the header!

And it shouldn't be too difficult to implement:

* You can add a new html template variable for the source link [here](https://github.com/Kotlin/dokka/blob/daed35f92b3b482688856d139da6849c8e6b4ab1/plugins/base/src/main/kotlin/renderers/html/innerTemplating/DefaultTemplateModelFactory.kt#L39).
* You can get the actual source link value from [the configuration](https://github.com/Kotlin/dokka/blob/fa22175d91fd63b833c537990b79569fa3e2f97d/core/src/main/kotlin/configuration.kt#L225), if it's present.
* The icon itself should be added to the default html template before the change theme icon [here](https://github.com/Kotlin/dokka/blob/393491918eb31fd1896c747e636965f917754b1b/plugins/base/src/main/resources/dokka/templates/includes/header.ftl#L20). For the link, you can use the variable added in the previous step

At the very least, an icon for GitHub should be added, but it'd be awesome to support Gitlab and Bitbucket as popular alternatives as well. 

I think in the beginning it'd be ok to NOT support linking to self-hosted websites, because it would probably require an additional configuration property or more complicated logic to choose the icon. So for now we can use the GitHub icon if `remoteUrl` begins with `https://github.com`. 

___

For anyone looking to get started with implementing this, see [Contributing Guidelines](https://github.com/Kotlin/dokka/blob/master/CONTRIBUTING.md) and [Developer Guides](https://kotlin.github.io/dokka/1.8.10/developer_guide/introduction/)

**Comment by aSemy** [comment]
#3235 only partially fixes this issue, right? It would be nice to support non-GitHub platforms.

**Comment by whyoleg** [comment]
@aSemy Sorry for confusion, details were changed in process of implementation of this feature, but the issue description was not updated. The last changes described in comment here: https://github.com/Kotlin/dokka/pull/3235#issuecomment-1783117290. (I've also now updated the description of the PR)

So in the end, direct link to homepage should be provided in configuration for `DokkaBase` plugin.

**Comment by aSemy** [comment]
Ah I see, thanks @whyoleg!

**Comment by IgnatBeresnev** [comment]
This feature has been released in 1.9.20 as experimental / hidden, as we are redesigning parts of Dokka's HTML format, and the homepage link's design / place might also change. Once the updated design is released, this feature will be covered in the official documentation on kotlinlang.

However, you can still give it a go now and give us feedback :)

HTML format's configuration (`DokkaBaseConfiguration`) has a new String property called `homepageLink`. You can set it to your project's homepage URL, like `https://github.com/kotlin/dokka`. Read this section in the docs to learn how to do it: https://kotlinlang.org/docs/dokka-html.html#configuration

Note: if you have a multi-project build, make sure to apply it to `dokkaHtmlPartial` as well as `dokkaHtmlMultiModule` tasks. The easiest way to do it is to configure the `AbstractDokkaTask` task.

The homepage icon is also overrideable, in case you want to use the GitHub icon instead of the default one. You can do it by passing a custom image named `homepage.svg` to the `customAssets` configuration option (square icons work best), read this section for more details: https://kotlinlang.org/docs/dokka-html.html#customize-assets
